### ISSUE TITLE
When "session restore" fails to establish connection ..

### ISSUE THREAD

**Issue by peterwy** [webyog/sqlyog-community#2179]
When "session restore" fails to establish connection we will creae a backup ysav.  It is possible to extract content (SQL, MXL) from this, but many users are not familiar with SQLite and have problems doing so. And also it takes to much time.

In ticket 92739 an idea came up, that if a connection cannot be restored a connection tab could/should still open named "connection_name - unconnected" or similar.

It may be after the .ysav has been saved, that SQLyog prompts user "do you want to see the saved content of unconnected servers?"

**Comment by peterwy** [comment]
se also https://github.com/webyog/sqlyog-community/issues/2120

**Comment by peterwy** [comment]
it is possible to extract content (SQL, MXL) from this
>>
it is possible to extract content (SQL, XML) from this

**Comment by peterwy** [comment]
.. and quite a lot of menus would then need to "grey out" as will (most) icons for an unconnected session.

**Comment by EagleEyeJohn** [comment]
## What's important to a user?
The SQL that they've spent time creating and testing.

## When restarting an SQLyog session, what does a user expect to see?
All of the connection and query tabs restored, which contain this valuable work.


## Current issues
When SQLyog currently restarts a session, any connection that it cannot reestablish is dropped from the session, along with all associated query tabs.

This results on a loss of previous effort, and a loss of state of where within a long process of many steps that you might have been.

What a user most definitely does not want is to be told that if they spend a lot of time picking their way through an SQLlite database they may get back some of what SQLyog has decided they no longer want.

## Why?
Because SQLyog cannot connect to display a list of databases.

## Does the user care?
NO! So long as they're told that the reason is because of there being no connection.

## Consider all of the following

- If a user starts SQLyog whilst travelling to/from the office and there's no connection to non-localhost hosts, so what... just restore the tabs and let the user connect when they get home.
- If a user starts SQLyog whilst in the office and all tabs were restored, and then leaves the office... SQLyog is still unable to connect to non-localhost hosts, and it is unable to navigate the schema tree.  How is that so different to the previous scenario?
- If a user starts SQLyog whilst in the office and all tabs were restored, and then works for a few hours amongst colleagues.  He/She or any of their colleagues may have modified schema and SQLyog is blissfully unaware of the changes and continues to display incorrect schema.  Being 100% accurate then doesn't seem to bother SQLyog.


So... please give control to the user, your customer, when dealing with unavailable hosts during session restore.

**Comment by greentree8** [comment]
This would be a great feature.  Would save me from having to come up with all these kludges to save temporary queries i'm working on.

**Comment by mclaborn** [comment]
Yes please! I often work in different locations and some of the servers I've connected to are not available. Those don't open and I lose whatever work was there for that server .

**Comment by LiquidisedFish** [comment]
@peterwy  - it is possible to extract content (SQL, MXL) from this
Could someone help with extracting queries from the backup ysav session

**Comment by LiquidisedFish** [comment]
> it is possible to extract content (SQL, MXL) from this
> 
> > >
> 
> it is possible to extract content (SQL, XML) from this

Would you please share the steps in order to get the contents
